awsService/src/main/java/com/kovan/app/service/S3Service.java

```diff
@@ -0,0 +1,128 @@
+import com.amazonaws.services.s3.model.*;
+import com.amazonaws.util.IOUtils;
+import com.kovan.entity.Document;
+import com.kovan.service.DocumentService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import static java.util.Objects.requireNonNull;
+
+@Service
+@Slf4j
+public class S3Service {
+
+    @Value("${bucketName}")
+    private String bucketName;
+
+    private final AmazonS3 s3Client;
+    private final DocumentService service;
+
+    public S3Service(AmazonS3 s3Client, DocumentService service) {
+        this.s3Client = s3Client;
+        this.service = service;
+    }
+
+    public String uploadFile(MultipartFile file) {
+
+        File tempFile = convertMultipartToFile(file);
+        String fileTypeFolder = determineFolder(file.getOriginalFilename());
+        String path = fileTypeFolder + "/" + file.getOriginalFilename();
+
+        String uniqueId = UUID.randomUUID().toString();
+
+        service.saveDocument(Document.builder()
+                .id(uniqueId)
+                .fileName(file.getOriginalFilename()).build());
+        try {
+            log.info("Uploading file {} to bucket {}, key: {}", file.getOriginalFilename(), bucketName, path);
+            s3Client.putObject(bucketName, path, tempFile);
+        } catch (Exception e) {
+            log.error("Error uploading file to S3", e);
+            throw new RuntimeException("File upload failed");
+        } finally {
+            if (!tempFile.delete()) {
+                log.warn("Temporary file {} could not be deleted", tempFile.getName());
+            }
+        }
+        return uniqueId;
+    }
+
+    private String determineFolder(String fileName) {
+
+        String fileExtension = getFileExtension(fileName);
+        return switch (fileExtension.toLowerCase()) {
+            case "pdf" -> "pdfs";
+            case "doc", "docx" -> "docs";
+            case "jpg", "jpeg", "png" -> "images";
+            case "txt" -> "texts";
+            default -> "others";
+        };
+    }
+
+    private String getFileExtension(String fileName) {
+        if (fileName == null || !fileName.contains(".")) {
+            throw new IllegalArgumentException("Invalid file name: " + fileName);
+        }
+        return fileName.substring(fileName.lastIndexOf('.') + 1);
+    }
+
+    private File convertMultipartToFile(MultipartFile file) {
+        File convertedFile = new File(requireNonNull(file.getOriginalFilename()));
+        try(FileOutputStream fos = new FileOutputStream(convertedFile)){
+            fos.write(file.getBytes());
+        }catch (IOException e) {
+           log.error("Error converting multipartFile to file", e);
```

Comment:
This catch won't serve the purpose of it.

Reply:
changed try catch method